File: src/main/java/com/teamwizardry/librarianlib/api/gui/components/mixin/ScissorMixin.java
package com.teamwizardry.librarianlib.api.gui.components.mixin;

import com.teamwizardry.librarianlib.api.gui.GuiComponent;
import com.teamwizardry.librarianlib.api.util.gui.ScissorUtil;
import com.teamwizardry.librarianlib.math.Vec2;

public class ScissorMixin {

	public static void scissor(GuiComponent<?> component) {
		component.preDraw.add( (c, pos, partialTicks) -> {
			Vec2 root = c.rootPos(new Vec2(0,0));
			ScissorUtil.push();
			ScissorUtil.set(root.xi, root.yi, c.getSize().xi, c.getSize().yi);
	        ScissorUtil.enable();
		});
		component.postDraw.addFirst( (c, pos, partialTicks) -> {
			ScissorUtil.pop();
		});
	}
	
}

File: modules/gui/src/main/java/com/teamwizardry/librarianlib/gui/mixin/ScissorMixin.kt
package com.teamwizardry.librarianlib.gui.mixin

import com.teamwizardry.librarianlib.gui.component.GuiComponent

object ScissorMixin {

    @Deprecated("Use ", replaceWith = ReplaceWith("component.clipping.clipToBounds = true"))
    fun scissor(component: GuiComponent) {
        component.clipToBounds = true
    }

}
